incountry/crypto_utils.py

```diff
@@ -10,83 +13,91 @@
 
 
 KEYS_TO_HASH = SERVICE_KEYS + SEARCH_KEYS
+EXTRA_KEYS_TO_ENCRYPT = ["precommit_body"]
 
 KEYS_TO_OMIT_ON_ENCRYPTION = ["created_at", "updated_at"]
 
-RESERVED_KEYS_TO_SEND = ["version", "is_encrypted"]
+RESERVED_KEYS_TO_SEND = ["body", "version", "is_encrypted"]
 
 
-def validate_crypto(crypto):
+def validate_crypto(crypto: InCrypto):
     if not isinstance(crypto, InCrypto):
         raise StorageClientException(f"'crypto' argument should be an instance of InCrypto. Got {type(crypto)}")
 
 
-def validate_is_string(value, arg_name):
+def validate_is_string(value: str, arg_name: str):
     if not isinstance(value, str):
         raise StorageClientException(f"'{arg_name}' argument should be of type string. Got {type(value)}")
 
 
-def is_json(data):
+def is_json(data: str) -> bool:
     try:
         json.loads(data)
     except ValueError:
         return False
     return True
 
 
-def hash(data):
+def hash(data: str) -> str:
     return hashlib.sha256(data.encode("utf-8")).hexdigest()
 
 
-def normalize_key(key, normalize=False):
+def normalize_key(key: Any, normalize: bool = False) -> Any:
```

Comment:
The type is `Any` but we could specify that the type remains the same like this:

```suggestion
from typing import TypeVar

T = TypeVar("T")

def normalize_key(key: T, normalize: bool = False) -> T:
```

Type checkers understand such notation and infer types better. So if you pass a string there, they understand that you recieve a string back.

Docs: https://docs.python.org/3/library/typing.html#typing.TypeVar

Reply:
The same approach could be applied to some other functions below.